File: AngularWebApiSample/src/AngularWebApiSample/ClientApp/src/app/api/models/CombinedResultModel.ts


import { ComplexBaseModel } from './ComplexBaseModel';

export interface ICombinedResultModel {
    $type?: string;
        id?: number;
        sampleList?: ComplexBaseModel[];
}

export class CombinedResultModel implements ICombinedResultModel {
    public $type: string;
        public id: number;
        public sampleList: ComplexBaseModel[];

    constructor(initObj?: ICombinedResultModel) {
        this.$type = 'AngularWebApiSample.Models.CombinedResultModel, AngularWebApiSample';
        if (initObj) {
            this.id = initObj.id || 0;
            this.sampleList = initObj.sampleList || [];
        } else {
            this.id = 0;
            this.sampleList = [];
        }
    }
}

File: src/AngularWebApiSample/Models/CombinedResultModel.cs
using System.Collections.Generic;
using AngularWebApiSample.Attributes;

namespace AngularWebApiSample.Models;

[GenerateFrontendType]
public class CombinedResultModel
{
    public int Id { get; set; }

#pragma warning disable MA0016 // Prefer return collection abstraction instead of implementation
    public List<ComplexBaseModel> SampleList { get; internal set; } = new List<ComplexBaseModel>();
#pragma warning restore MA0016 // Prefer return collection abstraction instead of implementation
}
